pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -187,3 +234,110 @@ func getPacketInfo(pktIn *ofctrl.PacketIn, ob *logInfo) error {
 	}
 	return nil
 }
+
+// rejectRequest sends reject response to the requesting client, based on the
+// packet-in message.
+func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
+	// Get ethernet data.
+	srcMAC := pktIn.Data.HWDst
+	dstMAC := pktIn.Data.HWSrc
+
+	var (
+		srcIP  string
+		dstIP  string
+		prot   uint8
+		isIPv6 bool
+	)
+	switch ipPkt := pktIn.Data.Data.(type) {
+	case *protocol.IPv4:
+		// Get IP data.
+		srcIP = ipPkt.NWDst.String()
+		dstIP = ipPkt.NWSrc.String()
+		prot = ipPkt.Protocol
+		isIPv6 = false
+	case *protocol.IPv6:
+		// Get IP data.
+		srcIP = ipPkt.NWDst.String()
+		dstIP = ipPkt.NWSrc.String()
+		prot = ipPkt.NextHeader
+		isIPv6 = true
+	}
+
+	// Get the OpenFlow ports of src Pod and dst Pod.
+	sIface, srcFound := c.ifaceStore.GetInterfaceByIP(srcIP)
+	dIface, dstFound := c.ifaceStore.GetInterfaceByIP(dstIP)
```

Comment:
I think @tnqn already left the same comment, but it seems like it wasn't addressed. `c.ifaceStore.GetInterfaceByIP` will only work if the IP address corresponds to a local Pod.
For ingress rules, only `dstIP` is guaranteed to match a local Pod.
For egress rules, only `srcIP` is guaranteed to match a local Pod.

I think you need to get inspiration from the Agent Traceflow code here. As Quan pointed out, there should be no need to set the output port, the pipeline should be able to compute the egress port.

Reply:
BTW, I am hoping that the e2e tests would fail because of this. There should be a good probability of Pods being scheduled on different Nodes and the reject message not being generated correctly. If the e2e tests always pass, we need to look into improving them.

Reply:
Changed.